src/components/Global/AddressLink/index.tsx

```diff
@@ -1,27 +1,37 @@
 import Link from 'next/link'
 import { useEffect, useState } from 'react'
-
+import { useEnsName } from 'wagmi'
+import { isAddress } from 'viem'
 import * as utils from '@/utils'
 
 const AddressLink = ({ address }: { address: string }) => {
     const [url, setUrl] = useState<string>('')
+    const [displayAddress, setDisplayAddress] = useState<string>(utils.printableAddress(address))
+
+    // Look up ENS name for any valid Ethereum address
+    const { data: ensName } = useEnsName({
+        address: isAddress(address) ? (address as `0x${string}`) : undefined,
+        chainId: 1, // Mainnet for ENS lookups
+    })
+
     useEffect(() => {
-        if (!address) return
-        if (address.endsWith('.eth')) {
-            utils.resolveFromEnsName(address).then((resolvedAddress) => {
-                if (!resolvedAddress) return
-                setUrl(`https://debank.com/profile/${resolvedAddress}`)
-            })
-            return
-        }
+        // Always set debank URL
         setUrl(`https://debank.com/profile/${address}`)
-    }, [address])
+
+        // Update display: prefer ENS name, fallback to shortened address
+        if (ensName) {
+            setDisplayAddress(ensName)
+        } else {
+            setDisplayAddress(utils.printableAddress(address))
+        }
+    }, [address, ensName])
```

Comment:
nit: I feel like there's no real need for effects or state here. We're just displaying conditionally either the ens is present or not

Reply:
no cuz we do:

0x --> fetch if ens exists --> ens